### PR TITLE
Adds v0.2.0-RC1 Changelog

### PR DESCRIPTION
**Pull request by danehans**
This should have been done with the RC1 release.

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

**Comment by arkodg** [comment]
rerunning this CI, raised https://github.com/envoyproxy/gateway/pull/469 which might solve the flake for 0.2.0

**Comment by danehans** [comment]
@arkodg @skriss this is passing CI.

File: changelogs/0.2.0-rc1.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  change: |
+    Added the EnvoyGateway API type for configuring Envoy Gateway.
+    Added the EnvoyProxy API type for configuring managed Envoys.
+- area: ci
+  change: |
+    Added tooling to build, run, etc. Envoy Gateway.
+- area: providers
+  change: |
+    Added the Kubernetes provider.
+- area: xds
```
Comment (arkodg) [inline changelogs/0.2.0-rc1.yaml:17]: nit: there are two separate sub components here - `xds-translator` and `xds-server`

**Review by arkodg — APPROVED** [review]
🚀

**Pull request merged**